src/main/frontend/src/components/shoppage/ShopCategories.js

```diff
@@ -0,0 +1,36 @@
+import classes from "./ShopCategories.module.css";
+import ShopCategoryItem from "./ShopCategoryItem";
+import { fetchCategories } from "../../utils/categoryService";
+import { useEffect, useState } from "react";
+
+function ShopCategories({categories, onCategoryChange, selected, selectedSuperCategory, onSuperCategoryChange}) {
+  const [categoriesFetched, setCategoriesFetched] = useState(false);
+
+  useEffect(async () => {
+    if (categories.length > 0) {
+      setCategoriesFetched(true);
+    }
+  }, [categories]);
+
+  const renderCategories = () => {
+    return (
+      <div>
+        {categories.filter(function (category) {
+            return category.supercategoryId == null;
+          })
+          .map((category) => (
+            <ShopCategoryItem onSuperCategoryChange = {onSuperCategoryChange} selectedSuperCategory = {selectedSuperCategory} selected={selected} onItemChange = {onCategoryChange} category = {category} />
+          ))}
```

Comment:
this can go directly in return method where you call this renderCategories(). There is no benefit of extracting it to separate method